### ISSUE TITLE
[web] If route name contains two slashes or more all sub path should exists if not an error will appear for each sub path

### ISSUE THREAD

**Issue by aragonet** [flutter/flutter#51319]
## Steps to Reproduce
```
void main() {
  Route<dynamic> _onGenerateRoute(RouteSettings settings) {
    print("Route settings $settings");
    var routes = <String, WidgetBuilder>{
      "/": (context) => Home(settings.arguments),
      "/day/details": (context) => DayDetailsPage(),
    };
    return MaterialPageRoute(
      settings: settings,
      builder: (ctx) => routes[settings.name](ctx),
    );
  }

  runApp(
    MaterialApp(
      initialRoute: '/',
      onGenerateRoute: _onGenerateRoute,
    ),
  );
}
```

Is you navigate to that route using `Navigator.pushNamed` function page loads correctly but if you try to refresh the page or directly navigate to that page, the following error appears.

I think the problem is that the navigator tries to load each part of the route path (as seen in logs) and  calls `routes[settings.name](ctx)`. If the sub path does not exists `routes[settings.name]` will be null and could not call the widget builder function.

<details>
  <summary>Logs</summary>

```
Route settings RouteSettings("/", null)
Route settings RouteSettings("/day", null)
Route settings RouteSettings("/day/details", null)
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following JSNoSuchMethodError was thrown building Builder(dirty):
NoSuchMethodError: tried to call a non-function, such as null: 'routes[$_get](...)'

The relevant error-causing widget was:
MaterialApp org-dartlang-app:///packages/Dailycheck/main.dart:35:5

When the exception was thrown, this was the stack:
package:Dailycheck/main.dart 30:40                                               <fn>
package:flutter/src/material/page.dart 82:27                                     buildPage
package:flutter/src/widgets/routes.dart 723:43                                   <fn>
package:flutter/src/widgets/basic.dart 6834:41                                   build
package:flutter/src/widgets/framework.dart 4397:28                               build
package:flutter/src/widgets/framework.dart 4329:15                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 5684:32                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 5575:14                               mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/framework.dart 4349:16                               performRebuild
package:flutter/src/widgets/framework.dart 4053:5                                rebuild
package:flutter/src/widgets/framework.dart 4312:5                                [_firstBuild]
package:flutter/src/widgets/framework.dart 4487:11                               [_firstBuild]
package:flutter/src/widgets/framework.dart 4307:5                                mount
package:flutter/src/widgets/framework.dart 3297:13                               inflateWidget
package:flutter/src/widgets/framework.dart 3091:12                               updateChild
package:flutter/src/widgets/binding.dart 1054:16                                 [_rebuild]
package:flutter/src/widgets/binding.dart 1025:5                                  mount
package:flutter/src/widgets/binding.dart 968:16                                  <fn>
package:flutter/src/widgets/framework.dart 2512:19                               buildScope
package:flutter/src/widgets/binding.dart 967:12                                  attachToRenderTree
package:flutter/src/widgets/binding.dart 847:24                                  attachRootWidget
package:flutter/src/widgets/binding.dart 830:7                                   <fn>
package:dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart 50:19  internalCallback

════════════════════════════════════════════════════════════════════════════════════════════════════
```

```
```

```
[✓] Flutter (Channel beta, v1.14.6, on Linux, locale es_ES.utf8)
    • Flutter version 1.14.6 at /flutter
    • Framework revision fabeb2a16f (4 weeks ago), 2020-01-28 07:56:51 -0800
    • Engine revision c4229bfbba
    • Dart version 2.8.0 (build 2.8.0-dev.5.0 fc3af737c7)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /development/android-sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /development/android-sdk
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.42.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.8.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.116
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!

```

</details>

**Comment by mdebbar** [comment]
@aragonet this isn't a bug. Flutter is working as intended. Let me explain.

When the page is refreshed, the `Navigator` will take the url as the `initialRoute`. Please take a look at the documentation for [`Navigator.initialRoute`](https://api.flutter.dev/flutter/widgets/Navigator/initialRoute.html). In your case, the navigator is parsing the path `/day/details` into parts (`/`, `/day`, `/day/details`) and tries to push each one of them. The problem happens when it tries to build the route for `/day`. The navigator calls `onGenerateRoute` that you provided, and it'll happily return a `MaterialPageRoute`. Then the navigator tries to build that page, and you get the exception because `routes[settings.name](ctx)` isn't a function.

Using [`onGenerateRoute`](https://api.flutter.dev/flutter/material/MaterialApp/onGenerateRoute.html) instead of the [`routes`](https://api.flutter.dev/flutter/material/MaterialApp/routes.html) hashmap, you are basically telling Flutter that you want to handle **_all_** routes. And your implementation of `onGenerateRoute` always returns a `MaterialPageRoute` which tells Flutter that the route is valid. In fact, you'll get the same error if you open a new window and go to `http://localhost:[port]/#/foo`.

TL;DR: to fix this issue, you'll need to return `null` from `_onGenerateRoute` when the given route name isn't valid.

```dart
void main() {
  Route<dynamic> _onGenerateRoute(RouteSettings settings) {
    print("Route settings $settings");
    var routes = <String, WidgetBuilder>{
      "/": (context) => Home(settings.arguments),
      "/day/details": (context) => DayDetailsPage(),
    };

    if (routes.containsKey(settings.name)) {
      return MaterialPageRoute(
        settings: settings,
        builder: (ctx) => routes[settings.name](ctx),
      );
    }

    return null;
  }

  runApp(
    MaterialApp(
      initialRoute: '/',
      onGenerateRoute: _onGenerateRoute,
    ),
  );
}
```

Tagging this as a good StackOverflow question.

**Comment by aragonet** [comment]
Thank you for your explanation! :100:
